tests/test_ec2.py

```diff
@@ -555,6 +555,23 @@ def test_ec2_snapshot_copy_tags(self):
         resources = policy.run()
         self.assertEqual(len(resources), 1)
 
+class TestAssociateInstanceProfile(BaseTest):
+
+    def test_ec2_associate_instance_profile(self):
+        session_factory = self.replay_flight_data(
+            'test_ec2_associate_instance_profile')
+        policy = self.load_policy({
+            'name': 'ec2-test-associate-instance-profile',
+            'resource': 'ec2',
+            'filters': [
+                {'tag:Name': 'MissingInstanceProfile'}],
+            'actions': [
+                {'type': 'associate-instance-profile',
+                 'name': 'ec2-default'}]},
+            session_factory=session_factory)
+        resources = policy.run()
+        self.assertEqual(len(resources), 1)
```

Comment:
we've got a few tests that have some bad verifications...

to be a bit clearer about the test assertion, can you add a filter to the test policy that matches that the instance doesn't have an instance profile (or different) and then use an ec2 client describe_instances to verify that the role is present on the instance after the policy execution.